rsocket/internal/SetupResumeAcceptor.cpp

```diff
@@ -143,24 +164,23 @@ void SetupResumeAcceptor::processFrame(
         break;
       }
 
-      if (!onResume(transport, std::move(resumeParams))) {
+      try {
+        onResume(transport, std::move(resumeParams));
+      } catch (RSocketException ex) {
         transport->outputFrameOrEnqueue(frameSerializer->serializeOut(
-            Frame_ERROR::rejectedResume("Resumption Failure")));
+            Frame_ERROR::rejectedResume(ex.what())));
```

Comment:
Discussed about this offline. The current design allows the server-application to pass any error message to the client in the ERROR msg payload (though not in a structured process).  This can be re-visited later independent of the API changes in the PR.